src/clients/python/library/tritonclient/http/__init__.py

```diff
@@ -1402,7 +1402,11 @@ def set_data_from_numpy(self, input_tensor, binary_data=True):
         else:
             self._data = None
             if self._datatype == "BYTES":
-                self._raw_data = serialize_byte_tensor(input_tensor).tobytes()
+                serialized_output = serialize_byte_tensor(input_tensor)
+                if serialized_output.size > 0:
```

Comment:
We already have a check in `serialize_byte_tensor()`:
```
    if input_tensor.size == 0:
        return np.empty([0])
```
Why not just modify that as needed?

Reply:
`serialize_byte_tensor` is used in other places too. That's why I had to do it here.